Lombiq.Hosting.Azure.ApplicationInsights.csproj

```diff
@@ -23,6 +23,7 @@
     <PackageReference Include="OrchardCore.ContentTypes.Abstractions" Version="1.2.2" />
     <PackageReference Include="OrchardCore.DisplayManagement" Version="1.2.2" />
     <PackageReference Include="OrchardCore.ResourceManagement" Version="1.2.2" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="6.0.0" />

```

Comment:
This looks like something we'd need to maintain together with AI. Why is it necessary, exactly?

Reply:
Beause of this 
![image](https://user-images.githubusercontent.com/19938236/154481550-09ad17a6-6284-4738-a6d7-f000161b4f85.png)

Reply:
Before this reference in the csproj this project didn't reference that package so I don't understand how this error can be produced before that (see it's last paragraph). Where are exactly the two conflicting references coming from before this change?

Reply:
It is not referenced directly, it is used by Microsoft.ApplicationInsights.AspNetCore.

Reply:
The [latest AI package](https://www.nuget.org/packages/Microsoft.ApplicationInsights.AspNetCore) doesn't depend on DiagnosticSource. What if you update it instead?

Reply:
Still, let's try to update to it because then we don't have to maintain this package reference here separately.

Reply:
What's the conflict there, exactly (not the error message but what you figured out)? It's including 6.0, what's including 5.0?

Reply:
Tbh I didn't find any which includes 5.0. I found this tho, but it's not 5.0, which makes this even weirder.
![image](https://user-images.githubusercontent.com/19938236/154691694-2f01747b-d780-48d2-9f3c-67f29a1b25ed.png)

Reply:
Please figure out where it's coming from. You might need to do a git clean at least. So, the ultimate question is, why is there a version mismatch and what should we do about it. Directly including the dependency of one of our dependencies doesn't seem like a good long-term solution.

Reply:
Maybe @barthamark could help with this, because I don't find anything else in the project and he got this problem too. He might found something else.

The only mismatch is between those that the error says. In `C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\5.0.0.......`

Reply:
Let's leave it then but together with the AI package update. Also, please open an issue here about removing this after the .NET 6 upgrade, link it from https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/issues/30, and open a PR for this change in OSOCE too and test it so we can see it doesn't break with .NET Core 3.1.

Reply:
Did it. can this be merged then?

Reply:
As for every such change, open an OSOCE PR with this submodule change as well: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/#contributing-and-support If it's looking good, we can merge.

Also, link https://github.com/Lombiq/Orchard-Azure-Application-Insights/issues/27 from a comment above the package reference so when just looking at the code we'll also know what's happening.

Reply:
Already opened a PR in OSOCE: https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/53
I'll fix the merge conflict.